tests/rados/test_scrub_parameters.py

```diff
@@ -0,0 +1,204 @@
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    1. Test to the forced flag during the scrubbing and deep-scrubbing
+    2. Test the scheduled time <=  osd_scrub_min_interval+
+        (osd_scrub_min_interval * osd_scrub_interval_randomized_ratio)
+    Returns:
+        1 -> Fail, 0 -> Pass
+    """
+
+    log.info(run.__doc__)
+    config = kw["config"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_object = RadosOrchestrator(node=cephadm)
+    scrub_object = RadosScrubber(node=cephadm)
+    mon_obj = MonConfigMethods(rados_obj=rados_object)
+
+    try:
+        for val in ("scrub", "deep-scrub"):
+            replicated_config = config.get("replicated_pool")
+            pool_name = replicated_config["pool_name"]
+            selected_osd = create_pool_get_osd(rados_object, pool_name)
+            dump_scrub_before_scrub = scrub_object.get_dump_scrubs(selected_osd)
+            if val == "scrub":
+                rados_object.run_scrub(osd=selected_osd)
+            else:
+                rados_object.run_deep_scrub(osd=selected_osd)
+            status, pg_id = chk_forced_flag(scrub_object, selected_osd)
+            if not status:
+                log.error(f"After the {val} none of the pgid is set as true")
+                return 1
+            forced_flag = wait_pg_forced_flag(scrub_object, selected_osd, pg_id)
+            if not forced_flag:
+                log.error(
+                    "The forced flag is not set to True and the waiting time is 20 minutes"
+                )
+                return 1
+            log.info(f"===Scenario-1: Completed by {val}bing the osd===")
+            pg_dump_scrub = scrub_object.get_pg_dump_scrub(selected_osd, pg_id)
+            pg_current_sched_time = pg_dump_scrub["sched_time"]
+            # Get the dump_scrub of the PG id
+            scheduled_time = verify_scheduled_time(
+                mon_object=mon_obj,
+                prev_dump_scrub=dump_scrub_before_scrub,
+                pg_id=pg_id,
+                pg_current_sched_time=pg_current_sched_time,
+            )
+            if not scheduled_time:
+                log.error("The scrub time is greater than upper bound time")
+                return 1
+            log.info(f"=====Verification completed by {val}bing the OSD====")
+            # deleted pool.This is due to execution time is taking more time until waiting for the cluster to clean
+            # state
+            method_should_succeed(rados_object.delete_pool, pool_name)
+            rados_object.log_cluster_health()
+            log.info(f"===Scenario-2:Completed by {val}bing the osd")
+    except Exception as e:
+        log.info(e)
+        log.info(traceback.format_exc())
+        return 1
+    finally:
+        log.info("Execution of finally block")
+        rados_object.log_cluster_health()
+    return 0
+
+
+def create_pool_get_osd(rados_object, pool_name):
+    """
+    Method is used -
+        1. Create replicated pool
```

Comment:
Are the workflows present in this test only Valid for Replicated pools?

If not I believe we should update the methods to accommodate both EC and RE pools for testing.

Reply:
I implemented the code that tests are executing on replicated and erasure pools.